src/Model/Item.php

```diff
@@ -2013,21 +2030,16 @@ public static function insert($item, $notify = false, $dontcache = false)
 	 * @return bool
 	 * @throws \Exception
 	 */
-	private static function insertActivity(&$item)
+	private static function insertActivity(array $item)
 	{
 		$activity_index = self::activityToIndex($item['verb']);
-
 		if ($activity_index < 0) {
-			return false;
+			// SHouldn't happen at all
```

Comment:
Then it probably should throw an Exception.

Reply:
I guess I prefer to remove this check completely, since technically it can't happen at all. The test before the function is called is based upon the same array that is used here.